batm_ssh_tunnel/src/main/java/com/generalbytes/batm/sshtunnel/Main.java

```diff
@@ -0,0 +1,52 @@
+/*************************************************************************************
+ * Copyright (C) 2014-2019 GENERAL BYTES s.r.o. All rights reserved.
+ *
+ * This software may be distributed and modified under the terms of the GNU
+ * General Public License version 2 (GPL2) as published by the Free Software
+ * Foundation and appearing in the file GPL2.TXT included in the packaging of
+ * this file. Please note that GPL2 Section 2[b] requires that all works based
+ * on this software must also be made publicly available under the terms of
+ * the GPL2 ("Copyleft").
+ *
+ * Contact information
+ * -------------------
+ *
+ * GENERAL BYTES s.r.o.
+ * Web      :  http://www.generalbytes.com
+ *
+ ************************************************************************************/
+package com.generalbytes.batm.sshtunnel;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+public class Main {
+
+    private static final File CONFIG_DIR = new File(System.getProperty("user.home"), ".batm_ssh_tunnel");
+
+    public static void main(String[] args) throws IOException, GeneralSecurityException {
+
+        if (!CONFIG_DIR.exists() && !CONFIG_DIR.mkdirs()) {
+            throw new IOException("Cannot create " + CONFIG_DIR.getAbsolutePath());
+        }
+
+
+        if (args.length > 0 && "init".equals(args[0])) {
+            String password = new Auth(CONFIG_DIR).init();
+            System.out.println(password); // intentionally stdout and not logging. The output is used in the install script
+
+        } else {
+            int port = Integer.parseInt(System.getProperty("batmSshTunnelPort", "22222"));
+            new SshServer(CONFIG_DIR, port).start();
+
```

Comment:
Write into console instructions for user what to do next. Go to CAS blabla...

Reply:
Instructions are displayed be the install script:

Adding user batmsshtunnel
Downloading and unpacking to /opt/batm_ssh_tunnel
Generating password
Installing the service
Created symlink /etc/systemd/system/multi-user.target.wants/batm-ssh-tunnel.service → /etc/systemd/system/batm-ssh-tunnel.service.
Starting the service

Please make the port 22222 is accessible from BATM Master Server.
No other ports need to be open.
Use this Tunnel Password in CAS Crypto Settings.
Make sure to keep the password SECRET!

********************************
Sh6f4MXb1&uRL%G2y3U-s%OR7EEIbJLq
********************************